Add support for multipart/form-data requests***

ziyacivan opened the pull request:
**Title:** Add Support for Multipart/Form-Data Requests
**Description**: I've encountered an issue when Silk attempts to handle `multipart/form-data` requests, specifically
when trying to access `request.body` directly. This operation triggers a `RawPostDataException` because Django consumes the stream upon parsing the multipart data, making it inaccessible afterwards.

To address this, I've implemented a check for `multipart/form-data` content type within the request handling logic. When such a content type is detected, the code now properly handles form data and file information using `request.POST` and `request.FILES`, avoiding direct access to `request.body`. This allows Silk to gracefully handle and log multipart requests without raising exceptions.

ziyacivan committed:
Add support for multipart/form-data requests***

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ziyacivan committed:
Refactor handling of multipart/form-data in RequestModelFactory

ziyacivan committed:
Merge remote-tracking branch 'refs/remotes/origin/add_multipart_support' into add_multipart_support

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ziyacivan committed:
Fix content type typo in model_factory.py

ziyacivan committed:
Merge branch 'add_multipart_support' of https://github.com/ziyacivan/django-silk into add_multipart_support

50-Course commented on silk/model_factory.py:
```diff
@@ -226,6 +226,16 @@ def construct_request_model(self):
         query_params = self.query_params()
         path = self.request.path
         view_name = self.view_name()
+        content_type = self.request.content_type
+
+        if content_type.startswith('multipart/form-data'):
+            body = {
+                'form_data': self.request.POST.dict(),
+                'files': {f: self.request.FILES[f].name for f in self.request.FILES}
+            }
+            raw_body = None  # Raw body is not available for multipart/form-data
+        else:
+            body, raw_body = self.body()
```
@ziyacivan Good catch, and thank you for making this patch! 👏🏼

ziyacivan replied:
Oh thank you so much! I couldn't continue to the development process (means is tests) because of my social life issues, thank you again 👏🏽

50-Course committed:
test(multipart_form_data): test form request creation works correctly

This test-patch addresses the newly updated patch changes when dealing
with `multipart/form_data` request

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

50-Course commented:
Just found out six different test are failing, I do have to fix them by tomorrow.
By the way, you are welcome to update the test cases if you want to.

Failure is related to implementation, not the test - most likely as a result of writing
implementation before writing the test cases.
